### PR TITLE
673 enhanced admin management

### PR DESCRIPTION
**Pull request by fozziethebeat**
Applies to #673 

Updates the admin views to user the backend's User API routes:
- Fetch all users from the backend and merge them with what the web knows
- Update web API routes to use the more updated backend routes
- Set the user's enabled and notes fields when managing a user

What's not supported yet:
- Pagination.  This is broken but will be fixed in a follow up PR.  How pagination works has changed when using the backend strategy.

### CODE REVIEW

File: website/src/pages/api/admin/user_messages.ts [hunk 1]
```diff
@@ -1,15 +1,12 @@
...
   const { user } = req.query;
-
-  const messagesRes = await fetch(`${process.env.FASTAPI_URL}/api/v1/frontend_users/local/${user}/messages`, {
-    method: "GET",
-    headers: {
-      "X-API-Key": process.env.FASTAPI_KEY,
-    },
-  });
-  const messages = await messagesRes.json();
+  const messages = await oasstApiClient.fetch_user_messages(user as string);
```
Comment (AbdBarho) [inline website/src/pages/api/admin/user_messages.ts:9]: string array? or `Message[]`?

File: website/src/pages/api/admin/users.ts [hunk 2]
```diff
@@ -1,31 +1,44 @@
...
+    result.set(user.id, user.role);
+    return result;
+  }, new Map());
+
+  const users = all_users.map((user) => {
+    const role = local_user_map.get(user.id) || "general";
+    return {
+      ...user,
+      role,
+    };
```
Comment (AbdBarho) [inline website/src/pages/api/admin/users.ts:41]: will this happen every time you open the users dashboard?

File: website/src/pages/api/admin/user_messages.ts (hunk 1 above)
Comment (fozziethebeat) [reply website/src/pages/api/admin/user_messages.ts]: Added the explicit type.

File: website/src/pages/api/admin/users.ts (hunk 2 above)
Comment (fozziethebeat) [reply website/src/pages/api/admin/users.ts:41]: Yes, it's not ideal and not something I want to keep for long.  I think the real fix is to push the role information to the backend but that would require a variety of new routes to make that fetching and setting easier.

**Review by AbdBarho — APPROVED** [review]

**Review by andreaskoepf — APPROVED** [review]
Front-end change approved!

**Review by yk — APPROVED** [review]
lgtm, thank you

**Comment by fozziethebeat** [comment]
Merging now that I see approval from all required people.

**Pull request merged**